### DISCUSSION TITLE
One OD object for many nodes?

### DISCUSSION THREAD

**Discussion by rcd09** [canopen-python/canopen#344] [General]
I have dozens of nodes that share an EDS.  Can I safely create one ObjectDictionary object and pass it to all the objects?  From what I can see the OD is read only but I want to make sure.

Thanks!

**Comment by acolomb** [comment]
Yes, that is perfectly fine, at least for `RemoteNode` objects. I don't know how the local node classes interact with it, but it doesn't make much sense to create several of those with a common OD usually.

What I do is have a class that derives from `RemoteNode`, with the OD of the specific device model as a class variable loaded once during the class definition. Each instance then refers to the common object by passing it to the parent class constructor during initialization.

**Comment by rcd09** [comment]
Thank you Andre. I was thinking along the same lines. I'll discover nodes
on the network with a network class and then create od objects as needed as
it instantiates each node, passing the od to the constructor as you mention.

Thanks!
